### ISSUE TITLE
Misleading error when TypeSpec can't be compiled

### ISSUE THREAD

**Issue by mccoyp** [tjprescott/openapi-diff#7]
**tl;dr:** when compiling TypeSpec fails, the current error -- that no Swagger or TypeSpec files were found -- is misleading and obscures that a compile error was the issue.

I tried referencing TypeSpec from my local `azure-rest-api-specs` repo clone, but the `diff` command kept failing with a "No Swagger or TypeSpec files found" error:
```bash
~/openapi-diff$ npm run diff -- --lhs KeyVaultOriginal --rhs ../azure-rest-api-specs/specification/keyvault/data-plane/Security.KeyVault.Secrets --compile-tsp --group-violations

> openapi-diff@1.0.0 diff
> node dist/src/index.js --lhs KeyVaultOriginal --rhs ../azure-rest-api-specs/specification/keyvault/data-plane/Security.KeyVault.Secrets --compile-tsp --group-violations

Unresolved references: ./examples/GetDeletedSecrets-example.json, ./examples/PurgeDeletedSecret-example.json, ./examples/GetDeletedSecret-example.json, ./examples/RecoverDeletedSecret-example.json, ./examples/GetSecrets-example.json, ./examples/RestoreSecret-example.json, ./examples/DeleteSecret-example.json, ./examples/SetSecret-example.json, ./examples/BackupSecret-example.json, ./examples/GetSecretVersions-example.json, ./examples/GetSecret-example.json, ./examples/UpdateSecret-example.json
Error: Command failed: tsp compile ../azure-rest-api-specs/specification/keyvault/data-plane/Security.KeyVault.Secrets --emit @azure-tools/typespec-autorest

    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: 'tsp compile ../azure-rest-api-specs/specification/keyvault/data-plane/Security.KeyVault.Secrets --emit @azure-tools/typespec-autorest'
}
file:///home/mcpatino/openapi-diff/dist/src/index.js:127
                    throw new Error(`No Swagger or TypeSpec files found: ${path}`);
                          ^

Error: No Swagger or TypeSpec files found: ../azure-rest-api-specs/specification/keyvault/data-plane/Security.KeyVault.Secrets
    at loadPaths (file:///home/mcpatino/openapi-diff/dist/src/index.js:127:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async main (file:///home/mcpatino/openapi-diff/dist/src/index.js:154:41)
    at async file:///home/mcpatino/openapi-diff/dist/src/index.js:44:1

Node.js v20.14.0
```

This made me think that I was somehow providing the folder path incorrectly, that file references outside the `openapi-diff` repository were invalid, or the target folder was improperly formatted in some way.

Fortunately the output shows the `tsp compile` command that's actually being run, so I was able to reproduce the compile error that caused this (which seems to have come from a compiler mismatch):
```bash
~/openapi-diff$ tsp compile ../azure-rest-api-specs/specification/keyvault/data-plane/Security.KeyVault.Secrets --emit @azure-tools/typespec-autorest
TypeSpec compiler v0.57.0

Diagnostics were reported during compilation:

warning compiler-version-mismatch: Current TypeSpec compiler conflicts with local version of @typespec/compiler referenced in /home/mcpatino/azure-rest-api-specs/specification/keyvault/data-plane/Security.KeyVault.Secrets. 
If this warning occurs on the command line, try running `typespec` with a working directory of /home/mcpatino/azure-rest-api-specs/specification/keyvault/data-plane/Security.KeyVault.Secrets. 
If this warning occurs in the IDE, try configuring the `tsp-server` path to /home/mcpatino/azure-rest-api-specs/node_modules/@typespec/compiler.
  Expected: /home/mcpatino/openapi-diff/node_modules/@typespec/compiler
  Resolved: /home/mcpatino/azure-rest-api-specs/node_modules/@typespec/compiler
/home/mcpatino/azure-rest-api-specs/node_modules/@azure-tools/typespec-azure-core/lib/models.tsp:266:56 - error reserved-identifier: Keyword cannot be used as identifier.
> 266 |   finalProperty?: TypeSpec.Reflection.ModelProperty | (valueof string);
      |                                                        ^^^^^^^
/home/mcpatino/azure-rest-api-specs/node_modules/@azure-tools/typespec-azure-core/lib/models.tsp:266:63 - error token-expected: ')' expected.
> 266 |   finalProperty?: TypeSpec.Reflection.ModelProperty | (valueof string);
      |                                                               ^
/home/mcpatino/azure-rest-api-specs/node_modules/@azure-tools/typespec-azure-core/lib/models.tsp:266:70 - error token-expected: ':' expected.
> 266 |   finalProperty?: TypeSpec.Reflection.ModelProperty | (valueof string);
      |                                                                      ^
/home/mcpatino/azure-rest-api-specs/node_modules/@azure-tools/typespec-azure-core/lib/models.tsp:267:1 - error token-expected: Statement expected.
> 267 | }
      | ^
/home/mcpatino/azure-rest-api-specs/node_modules/@typespec/http/lib/auth.tsp:114:80 - error unassignable: Type '[]' is not assignable to type 'valueof string[]'
> 114 | model OAuth2Auth<Flows extends OAuth2Flow[], Scopes extends valueof string[] = []> {
      |                                                                                ^^
/home/mcpatino/azure-rest-api-specs/node_modules/@typespec/http/lib/auth.tsp:122:18 - error value-in-type: Template parameter can be passed values but is used as a type.
> 122 |   defaultScopes: Scopes;
      |                  ^^^^^^
/home/mcpatino/azure-rest-api-specs/node_modules/@typespec/http/lib/auth.tsp:220:21 - error value-in-type: Template parameter can be passed values but is used as a type.
> 220 |   openIdConnectUrl: ConnectUrl;
      |                     ^^^^^^^^^^
/home/mcpatino/azure-rest-api-specs/node_modules/@azure-tools/typespec-azure-core/lib/models.tsp:257:18 - error value-in-type: Template parameter can be passed values but is used as a type.
> 257 |   finalProperty: FinalProperty;
      |                  ^^^^^^^^^^^^^
/home/mcpatino/azure-rest-api-specs/node_modules/@azure-tools/typespec-azure-core/lib/models.tsp:257:3 - error override-property-mismatch: Model has an inherited property named finalProperty of type FinalProperty which cannot override type ModelProperty | ErrorType
> 257 |   finalProperty: FinalProperty;
      |   ^^^^^^^^^^^^^

Found 9 errors, 1 warning.
```

When I moved the specs into the `openapi-diff` directory, the same command worked as expected.

**Comment by tjprescott** [comment]
Fixed in https://github.com/tjprescott/openapi-diff/commit/7999b1066f1e6dd6dff5bd1718a46d3099f48a58
